### ISSUE TITLE
1.8.0 - nginx won't start

### ISSUE THREAD

**Issue by mikeatdesk** [engintron/engintron#393]
Just upgraded to 1.8.0 and none of my domains will load.

Looks like nginx won't launch, and Engintron cites "Something went wrong, check the Nginx log.".

The log shows multiple lines of: `2017/02/17 13:10:31 [error] 15262#15262: *291 connect() failed (111: Connection refused) while connecting to upstream,`

And multiple lines of `2017/02/17 13:14:16 [emerg] 16484#16484: could not build server_names_hash, you should increase server_names_hash_bucket_size: 64`

Not sure how to fix it, and had to disable Engintron in the meantime.

**Comment by mikeatdesk** [comment]
I increase that limit after a bit of Googling and nginx still won't start.

nginx log now shows `2017/02/17 13:17:28 [emerg] 16900#16900: listen() to 0.0.0.0:443, backlog 511 failed (98: Address already in use)
2017/02/17 13:17:28 [emerg] 16900#16900: listen() to 0.0.0.0:443, backlog 511 failed (98: Address already in use)
2017/02/17 13:17:28 [emerg] 16900#16900: listen() to 0.0.0.0:443, backlog 511 failed (98: Address already in use)
2017/02/17 13:17:28 [emerg] 16900#16900: listen() to 0.0.0.0:443, backlog 511 failed (98: Address already in use)
2017/02/17 13:21:49 [emerg] 17881#17881: listen() to 0.0.0.0:443, backlog 511 failed (98: Address already in use)
2017/02/17 13:21:49 [emerg] 17881#17881: listen() to 0.0.0.0:443, backlog 511 failed (98: Address already in use)
2017/02/17 13:21:49 [emerg] 17881#17881: listen() to 0.0.0.0:443, backlog 511 failed (98: Address already in use)
2017/02/17 13:21:49 [emerg] 17881#17881: listen() to 0.0.0.0:443, backlog 511 failed (98: Address already in use)
2017/02/17 13:21:49 [emerg] 17881#17881: listen() to 0.0.0.0:443, backlog 511 failed (98: Address already in use)
2017/02/17 13:21:49 [emerg] 17881#17881: still could not bind()`

This same setup worked fine in 1.7.3, so I'm thinking 1.8.0 is buggy.

**Comment by mikeatdesk** [comment]
Got nginx to load, but it's serving the server's shared certs instead of each domain's custom certs, resulting in cert mismatch errors.

Not sure where to go to troubleshoot that, so will disable Engintron for now.

**Comment by fevangelou** [comment]
Besides that value bump, your issue was related to some stuck process from
either Nginx or Apache, which did not allow port 443 to be released for
Nginx to use it.

Try re-installing. Each domain will use the certificate it has assigned in
Apache.

On Feb 17, 2017 8:25 PM, "mikeatdesk" <notifications@github.com> wrote:

Got nginx to load, but it's serving the server's shared certs instead of
each domain's custom certs, resulting in cert mismatch errors.

Not sure where to go to troubleshoot that, so will disable Engintron for
now.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/engintron/engintron/issues/393#issuecomment-280727910>,
or mute the thread

**Comment by mikeatdesk** [comment]
I reinstalled, upped the server_names_hash_max_size and server_names_hash_bucket_size so nginx and apache both are running, but every domain threw 502 errors, so I disabled Engintron again.

**Comment by mikeatdesk** [comment]
Got it working after 5 installs, and 2 reboots.

**Comment by fevangelou** [comment]
Forgot to mention you must clear Nginx's caches. That was probably the
culprit. I'll automate this process when updating. Thanks for the precious
feedback!

**Comment by gspace1** [comment]
We have run into this same problem, we have had to already do reinstalls a few times on one of the servers because Nginx fails and won't restart.

We have installed this on several of our servers, the other problem we run into is clients can't access via SSL after install, the only solution we found is we switch the port back to 443 from 8443 so their sites will respond. The logs show a lot of conflicts after install on port 443.
